Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -44,33 +45,112 @@ public MigrationTool()
+        public bool TryAddObjectForMigration(Type type, Object selectedObject)
         {
+            if(type == null)
+            {
+                Debug.LogError("Migration type needs to be selected before migration.");
+                return false;
+            }
+
+            if(type != migrationHandlerInstanceType)
+            {
+                ClearMigrationList();
+                Debug.LogWarning("New migration type selected for migration. Clearing previous selection.");
+
+                if (!SetMigrationHandlerInstance(type))
+                {
+                    return false;
+                }
+            }
+
             if (!selectedObject)
             {
-                Debug.LogError("Selection is Empty. Could not add for migration.");
+                Debug.LogError("Selection is empty. Please select object for migration.");
                 return false;
             }
 
             if (selectedObject is GameObject || selectedObject is SceneAsset)
             {
-                if (!migrationObjects.Contains(selectedObject))
+                if (CheckIfCanMigrate(type, selectedObject) && !migrationObjects.Contains(selectedObject))
                 {
                     migrationObjects.Add(selectedObject);
-                    Debug.Log($"{selectedObject.name} object added for migration");
+                    return true;
+                }
+                else
+                {
+                    Debug.Log($"{selectedObject.name} does not support {type.Name} migration. Could not add object for migration");
+                    return false;
                 }
-                return true;
             }
             Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for migration");
             return false;
         }
 
+        private bool CheckIfCanMigrate(Type type, Object selectedObject)
+        {
+            string objectPath = AssetDatabase.GetAssetPath(selectedObject);
+
+            // Object migrationObject is a scene game object
+            if (String.IsNullOrEmpty(objectPath) && selectedObject is GameObject)
+            {
+                var objectHierarchy = ((GameObject)selectedObject).GetComponentsInChildren<Transform>();
+                for (int i = 0; i < objectHierarchy.Length; i++)
+                {
+                    try
+                    {
+                        if (migrationHandlerInstance.CanMigrate(objectHierarchy[i].gameObject))
+                        {
+                            return true;
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        Debug.LogError($"{e.Message}: Could not check if GameObject {objectHierarchy[i].name} can be migrated");
+                    }
+                }
+            }
+            else
+            {
+                // Selected object is prefab asset
+                if (selectedObject is GameObject)
```

Comment:
Nit: The nesting of these if/else statements is somewhat deep, where conditions can be maybe a little hard to decipher later. i.e. to be a bit clearer by what I mean by this, consider that this condition is nested in a case that validates that objectPath is null (and thus, this case is not null) - it gets a bit weirder in the else -> else if case down below (where it's even a bit harder to tell what the prexisting condtions are).

For readability going forward, consider having something like

if (IsSceneGameObject(...)) {
}
else if (IsPrefabAsset(...))
{
}
else if (IsSceneAsset(...))